compose/config/validation.py

```diff
@@ -162,10 +164,25 @@ def _parse_oneof_validator(error):
         Inspecting the context value of a ValidationError gives us information about
         which sub schema failed and which kind of error it is.
         """
+
+        required = [context for context in error.context if context.validator == 'required']
+        if required:
+            return required[0].message
+
+        additionalProperties = [context for context in error.context if context.validator == 'additionalProperties']
+        if additionalProperties:
+            invalid_config_key = _parse_key_from_error_msg(additionalProperties[0])
+            return "contains unsupported option: '{}'".format(invalid_config_key)
```

Comment:
It's providing better error messages for oneOf (tests.unit.config.config_test).

additionalProperties:
- test_config_invalid_environment_dict_key_raises_validation_error
- test_extends_validation_invalid_key

required:
- test_extends_validation_empty_dictionary
- test_extends_validation_missing_service_key